components/arrow_ext/src/ipc.rs

```diff
@@ -41,29 +42,82 @@ pub enum Compression {
 // The lower the level, the faster the speed (at the cost of compression).
 const ZSTD_LEVEL: i32 = 3;
 
-pub fn encode_record_batch(batch: &RecordBatch, compression: Compression) -> Result<Vec<u8>> {
-    let buffer: Vec<u8> = Vec::new();
-    let mut stream_writer = StreamWriter::try_new(buffer, &batch.schema()).context(ArrowError)?;
-    stream_writer.write(batch).context(ArrowError)?;
-    stream_writer
-        .into_inner()
-        .context(ArrowError)
-        .and_then(|bytes| match compression {
-            Compression::None => Ok(bytes),
+#[derive(Default)]
+/// Encoder that can encode a batch of record batches with specific compression
+/// method.
+pub struct RecordBatchesEncoder {
+    stream_writer: Option<StreamWriter<Vec<u8>>>,
+    compression: Compression,
+    num_rows: usize,
+}
+
+impl RecordBatchesEncoder {
+    pub fn new(compression: Compression) -> Self {
+        Self {
+            stream_writer: None,
+            compression,
+            num_rows: 0,
+        }
+    }
+
+    /// Get the number of rows that have been encoded.
+    pub fn num_rows(&self) -> usize {
+        self.num_rows
+    }
+
+    /// Append one batch into the encoder for encoding.
+    pub fn write(&mut self, batch: &RecordBatch) -> Result<()> {
+        let stream_writer = if let Some(v) = &mut self.stream_writer {
+            v
+        } else {
+            let buffer: Vec<u8> = Vec::new();
+            let stream_writer =
+                StreamWriter::try_new(buffer, &batch.schema()).context(ArrowError)?;
+            self.stream_writer = Some(stream_writer);
+            self.stream_writer.as_mut().unwrap()
+        };
+
+        stream_writer.write(batch).context(ArrowError)?;
+        self.num_rows += batch.num_rows();
+        Ok(())
+    }
+
+    /// Finish encoding and generate the final encoded bytes.
```

Comment:
```suggestion
    /// Finish encoding and generating the final encoded bytes.
```

Reply:
Actually, It reads `Finish encoding, and then generate ...`.